cli/src/cmd_auth.rs

```diff
@@ -111,240 +119,255 @@ pub fn parse_host(input: &str) -> Result<url::Url> {
 //     }
 // }
 
-/// Authenticate with an Oxide host.
-///
-/// Alternatively, pass in a token on standard input by using `--with-token`.
-///
-///     # start interactive setup
-///     $ oxide auth login
+/// Authenticate with an Oxide Silo
 ///
-///     # authenticate against a specific Oxide instance by reading the token
-///     # from a file
-///     $ oxide auth login --with-token --host oxide.internal < mytoken.txt
-///
-///     # authenticate with a specific Oxide instance
+///     # authenticate with a specific Oxide silo
 ///     $ oxide auth login --host oxide.internal
 ///
-///     # authenticate with an insecure Oxide instance (not recommended)
+///     # authenticate with an insecure Oxide silo (not recommended)
 ///     $ oxide auth login --host http://oxide.internal
 #[derive(Parser, Debug, Clone)]
 #[command(verbatim_doc_comment)]
 pub struct CmdAuthLogin {
-    /// Read token from standard input.
-    #[clap(long)]
-    with_token: bool,
-
     /// The host of the Oxide instance to authenticate with.
     /// This assumes https; specify an `http://` prefix if needed.
     #[clap(short = 'H', long, value_parser = parse_host)]
     host: url::Url,
 
     /// Override the default browser when opening the authentication URL.
-    #[clap(long, group = "browser-options")]
+    #[clap(long)]
     browser: Option<String>,
 
     /// Print the authentication URL rather than opening a browser window.
-    #[clap(long = "no-browser", group = "browser-options")]
+    #[clap(long)]
     no_browser: bool,
 }
 
 impl CmdAuthLogin {
-    pub async fn run(&self, ctx: &Context) -> Result<()> {
-        // if !ctx.io.can_prompt() && !self.with_token {
-        //     return Err(anyhow!(
-        //         "--with-token required when not running interactively"
-        //     ));
-        // }
-
-        let token = if self.with_token {
-            // Read the token from stdin.
-            let mut token = String::new();
-            std::io::stdin().read_to_string(&mut token)?;
-            token = token.trim_end_matches('\n').to_string();
-
-            if token.is_empty() {
-                bail!("token cannot be empty");
+    pub async fn login(&self, ctx: &Context) -> Result<()> {
+        let profile = ctx.client_config().profile();
+
+        if let Some(profile_name) = profile {
+            // If the profile already has a token, alert the user.
+            if ctx.cred_file().profile.contains_key(profile_name)
+                && !yes(format!(
+                    "The profile \"{}\" is already authenticated. {}",
+                    profile_name, "Do you want to re-authenticate?",
+                ))?
+            {
```

Comment:
Note that we haven't checked the validity of the credentials at this point. It may be that people want to re-authenticate because the existing creds are invalid (... for reasons?). Note that this is preserving existing functionality; if we want to change it, could we do so outside the scope of this PR?